pkg/convert/pprof/profile.go

```diff
@@ -210,7 +211,16 @@ func (p *RawProfile) Parse(ctx context.Context, putter storage.Putter, _ storage
 					p.parser = nil
 				}()
 			} else {
-				p.parser = streaming.NewStreamingParser(config)
+				if p.ArenasEnabled {
+					config.ArenasEnabled = true
+				}
+				parser := streaming.NewStreamingParser(config)
+				p.parser = parser
+				if p.ArenasEnabled {
+					defer func() {
+						parser.FreeArena()
+					}()
```

Comment:
Just a note:

My biggest concern is the lifetime of the trees we Put to OSS cache/storage. I'm almost sure that we reference the source object in some cases (e.g I wouldn't clone a tree when put it to a cache for the first time). This is not something that makes use of arenas for trees impossible, but if we are up to enabling them in OSS, we need to carefully analyse object ownership beforehand.

I think in future we will also need to add `tree.Free()` because currently you need to own the arena to do so, which is not always easy to achieve. For example, if we had this method and freed trees before evicting them from the cache, tree lifecycle would be pretty explicit. However, this also requires allocating an arena per tree.

Reply:
I believe we should not enable arenas for OSS. Arena is good for a short lived object with a well defined lifetime, but as you said a tree is kept in memory sometimes for unknown time span. It is hard reason about lifecycle of a tree. Also multiple trees will share a single arena so they will need to coordinate somehow in order to free when all of them are free which is error-prone and extends arena life time(increases memory).
Maybe we even should not introduce arena to a Tree at all - after "moving parsers" project we will not need a tree as an intermediate representation in most of cases.